ggshield/core/config/user_config.py

```diff
@@ -62,26 +57,44 @@ def load(cls, config_path: Optional[str] = None) -> Tuple["UserConfig", str]:
 
         user_config = UserConfig()
         if config_path:
-            data = load_yaml(config_path) or {}
-            user_config.update_config(data)
+            user_config._update_from_file(config_path)
             return user_config, config_path
 
         for global_config_filename in GLOBAL_CONFIG_FILENAMES:
             global_config_path = get_global_path(global_config_filename)
-            global_data = load_yaml(global_config_path)
-            if global_data and user_config.update_config(global_data):
+            if os.path.exists(global_config_path):
+                user_config._update_from_file(global_config_path)
                 break
```

Comment:
nit: not related to this PR but we could reorder the local and the global and return `user_config, config_path` instead of breaking, that would prevent having to run the global part if a local config is set

Reply:
Ha, I didn't realize the global config was still taken but only overriden for the fields which are defined in the local config. Disregard this comment

Reply:
Ha, I didn't realize the global config was still taken but only overriden for the fields which are defined in the local config. Disregard this comment